### ISSUE TITLE
Embedding pipeline: tokenize → forward → pool → normalize

### ISSUE THREAD

**Issue by anibjoshi** [strata-ai-labs/strata-inference#14]
### Context
This is the primary deliverable — a high-level API that takes text and returns embedding vectors. It orchestrates all prior milestones: GGUF loading, tokenization, model forward pass, and pooling.

### API Design
```rust
pub struct EmbeddingEngine {
    model: Model,
    tokenizer: Box<dyn Tokenizer>,
    config: ModelConfig,
    backend: Arc<dyn ComputeBackend>,
}

impl EmbeddingEngine {
    /// Load model from GGUF file
    pub fn load(path: &str) -> Result<Self>;

    /// Embed a single text
    pub fn embed(&self, text: &str) -> Result<Vec<f32>>;

    /// Embed a batch of texts (more efficient)
    pub fn embed_batch(&self, texts: &[&str]) -> Result<Vec<Vec<f32>>>;

    /// Get embedding dimension
    pub fn embedding_dim(&self) -> usize;
}
```

### Forward Pass Pipeline
```
1. Tokenize: text → token_ids + attention_mask
2. Token embedding lookup: token_ids → [seq_len, hidden_size]
3. Position embedding:
   - BERT: add learned position embeddings
   - Gemma: positions applied via RoPE in attention
4. Embedding normalization (if applicable)
5. N transformer layers
6. Final normalization
7. Pooling:
   - Mean: average over non-padding tokens
   - CLS: extract first token
   - Last: extract last token
8. L2 normalization (for cosine similarity)
```

### Batch Processing Strategy
- Pad all sequences in batch to max length
- Single batched forward pass through all layers
- Attention mask ensures padding doesn't affect results

### Acceptance Criteria
- [ ] `EmbeddingEngine::load(path)` — full pipeline from GGUF file
- [ ] `embed(text)` returns `Vec<f32>` of length `hidden_size`
- [ ] `embed_batch(texts)` processes batch efficiently
- [ ] Pooling strategy selected from GGUF metadata or config
- [ ] L2 normalization of output embeddings
- [ ] Cosine similarity between related texts > 0.7
- [ ] Match HuggingFace EmbeddingGemma outputs within tolerance (cosine sim > 0.99)
- [ ] Benchmark: throughput in tokens/sec and texts/sec
- [ ] Memory profiling: peak GPU memory for 300M Q8_0 model

### Target Model: EmbeddingGemma-300M-Q8_0
- 308M parameters, 24 layers, 768 hidden, 16 heads
- Q8_0 quantization → ~310MB file
- SentencePiece BPE tokenizer
- RMSNorm, SwiGLU, RoPE
- Mean pooling

### Estimated LOC: ~500-700
